src/main/java/com/google/sps/daos/RealtimeDao.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.daos;
+
+import java.io.IOException;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.google.sps.data.Counter;
+import java.io.IOException;
+
+public class RealtimeDao implements CounterDao {
+  
+  private Counter counter;
+
+  private DatabaseReference realtimeDb;
+
+  public RealtimeDao() {
+
+    try {
+      FirebaseOptions options = new FirebaseOptions.Builder()
```

Comment:
You might want to initialize Firebase in the ServletContextListener and then pass the app instance to this constructor. That way, any other DAOs in the future can reuse it.

I also think it will give you an error if you try to initialize Firebase more than once.

Reply:
Done.